dbms/src/DataStreams/LimitByBlockInputStream.cpp

```diff
@@ -0,0 +1,86 @@
+#include <DB/DataStreams/LimitByBlockInputStream.h>
+
+namespace DB
+{
+
+LimitByBlockInputStream::LimitByBlockInputStream(BlockInputStreamPtr input_, size_t value_, Names columns_)
+	: columns_names(columns_)
+	, value(value_)
+{
+	children.push_back(input_);
+}
+
+String LimitByBlockInputStream::getID() const
```

Comment:
The purpose of method `getID` - to allow merging common elements of few execution pipelines. See `glueBlockInputStreams`. So, method `getID` must return different values for different calculations. In this case, it is missing names and limit value.

It is experimental, not used in production. Despite of this, better to implement method `getID` correctly.